src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,158 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * Represent list of tasks entered by user
+ */
+public class TaskList {
+    /**
+     * Arraylist of Tasks
+     */
+    private final ArrayList<Task> actions = new ArrayList<>();
+
+    /**
+     * Adds Task to the TaskList
+     *
+     * @param type Type of task to be added
+     * @param inputs Array of parsed input from user
+     * @return Boolean representing success of add operation
+     * @throws EmptyDescriptionException if description is empty
+     */
+    public boolean add(TaskType type, String[] inputs) throws EmptyDescriptionException {
```

Comment:
Perhaps `addTask` would be a better name to describe what this method does, but if this name is clear enough in its context then it should be fine.